plugin/trino-hive/src/main/java/io/trino/plugin/hive/FileWriter.java

```diff
@@ -25,7 +25,10 @@
 
     void appendRows(Page dataPage);
 
-    void commit();
+    /**
+     * Commits written data. Returns {@link RollbackAction} which can be used to restore state before data was written.
```

Comment:
"restore state before data was written" is confusing. How about "cleanup on failure"